src/main/java/com/example/nlern/controllers/subloadController.java

```diff
@@ -1,23 +1,144 @@
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.List;
 
 @RestController
-@RequestMapping("/subload")
-public class subloadController {
+@CrossOrigin
+@RequestMapping("/subload" )
+public class SubloadController {
+
+    @Autowired
+    private QuestionRepository questionRepository;
+
     @Autowired
-    private GroupRepository GroupRepo;
-    @PostMapping("/check_group")
-    public Map<String, Object> createUser(@RequestParam("group") String group){
+    private TestRepository testRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
 
-        Map<String, Object> map = new HashMap<>();
-        map.put("groupExist", GroupRepo.existsByName(group));
-        return map;
+    @GetMapping("check_group/{name}" )
+    public boolean checkGroupName(@PathVariable  String name){
+        return groupRepository.existsByName(name);
     }
+
+    @GetMapping("teachersHome")
+    public List<Test> home(Model model){
+        int teacherId = 3; //TODO get id from auth
+        return  testRepository.findTop5ByAuthorIdOrderByIdDesc(teacherId);
+    }
+
+
+    @DeleteMapping("tests/{id}")
+    public ResponseEntity deleteClient(@PathVariable Integer id) {
+        questionRepository.deleteAllByTestId(id);
+        testRepository.deleteById(id);
+
+        return ResponseEntity.ok().build();
+    }
+
+
+    @PostMapping("tests")
+    public ResponseEntity createTest(@RequestBody FullTest fullTest) {
+
+        int teacherId = 3; //TODO get id from auth
+
+        Test test = fullTest.getTest();
+        List<Question> questions = fullTest.getQuestions();
+
+        test.setAuthor(userRepository.getById(teacherId));
+        test.setGroup(groupRepository.getByName(fullTest.getGroupName()));
+        test = testRepository.save(test);
+
+        Test finalTest = test;
+        questions.stream()
+                .forEach(q-> {
+                    q.setTest(finalTest);
+                    questionRepository.save(q);
+                });
+
+        return ResponseEntity.ok().build();
+    }
+
+    @PutMapping("tests/{id}")
+    public ResponseEntity updateTest(@PathVariable Integer id, @RequestBody FullTest fullTest) {
+        Test finalTest;
+        Test test = testRepository.findById(id).orElseThrow(RuntimeException::new);
+        Test receivedTest = fullTest.getTest();
+        List<Question> receivedQuestions = fullTest.getQuestions();
+
+        List<Integer> createdQuestions = fullTest.getAdded();
+        List<Integer> deletedQuestions = fullTest.getDeleted();
+
+        test.setName(receivedTest.getName());
+        test.setDuration(receivedTest.getDuration());
+        test.setGroup(groupRepository.getByName(fullTest.getGroupName()));
+        test = testRepository.save(test);
+        finalTest = test;
+
+
+        for (int i: deletedQuestions) { //TODO add if(exist)
+            questionRepository.deleteById(i);
```

Comment:
This unboxing into primitive will kill performance, rewrite using deletedQuestions.forEach

Reply:
Corrected